### ISSUE TITLE
Submodule not explicitly imported shows as valid in completions

### ISSUE THREAD

**Issue by MikhailArkhipov** [microsoft/pyright#496]
**Describe the bug**
If module A has a submodule B and A does not import B, B still shows up in completions while Python produces an error.

**To Reproduce**
Create project folder P
`P/app.py`: 
```python
import package
```
`P/package/__init__.py`: empty
`P/package/m1/__init__.py`: 
```python
x = 1
```
Open P and in `app.py` type 
```python
import package
package.
```

OBSERVED: `m1` shows up in completions
Try running code in Python replacing `package1.` with `print(package.m1.x)`:
```
>c:\Python\Python37\python.exe app.py
Traceback (most recent call last):
  File "app.py", line 2, in <module>
    print(package.m1.x)
AttributeError: module 'package' has no attribute 'm1'
```
Now change `package/__init__.py` into
```python
import package.m1 as m2
```
Run `app.py` again. Now it succeeds

**Expected behavior**
Module `m1` should not appear in completions when Python does not recognize it as a member of `package`. The problem is that completion offers user code that does not actually work.

Related: https://github.com/microsoft/pyright/issues/439

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next published version of Pyright.

Here's the commit, if you're interested.
https://github.com/microsoft/pyright/commit/25758b8f37faf296f80eb5fd73995dd94b3781f9

**Comment by erictraut** [comment]
This is fixed in 1.1.23, which I just published.
